lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -459,27 +465,35 @@ def success_from(action, response)
           response['response'] == "[#{action}-received]"
         when 'adjustAuthorisation'
           response['response'] == 'Authorised' || response['response'] == '[adjustAuthorisation-received]'
+        when 'storeToken'
+          response['result'] == 'Success'
         else
           false
         end
       end
 
       def message_from(action, response)
         return authorize_message_from(response) if action.to_s == 'authorise' || action.to_s == 'authorise3d'
-        response['response'] || response['message']
+        response['response'] || response['message'] || response['result']
       end
 
       def authorize_message_from(response)
         if response['refusalReason'] && response['additionalData'] && response['additionalData']['refusalReasonRaw']
           "#{response['refusalReason']} | #{response['additionalData']['refusalReasonRaw']}"
         else
-          response['refusalReason'] || response['resultCode'] || response['message']
+          response['refusalReason'] || response['resultCode'] || response['message'] || response['result']
         end
       end
 
       def authorization_from(action, parameters, response)
         return nil if response['pspReference'].nil?
-        recurring = response['additionalData']['recurring.recurringDetailReference'] if response['additionalData']
+
+        if action == 'storeToken'
+          recurring = response['recurringDetailReference']
+        else
+          recurring = response['additionalData']['recurring.recurringDetailReference'] if response['additionalData']
+        end
+
```

Comment:
The `if` on line 494 is triggering a Rubocop error because it is inside an `if/else` block. Variables with a ternary operator is one solution.

```
recurring = response['additionalData']['recurring.recurringDetailReference'] if response['additionalData']
action == 'storeToken' ? recurring = response['recurringDetailReference'] : recurring = recurring
```

Reply:
I like DeeDee's suggestion of reversing the order (maybe a simple conditional would suffice then).

```ruby
recurring = response['additionalData']['recurring.recurringDetailReference'] if response['additionalData']
recurring = response['recurringDetailReference'] if action == 'storeToken'
```